pulsar-client/src/main/java/com/yahoo/pulsar/client/api/ConsumerConfiguration.java

```diff
@@ -171,4 +177,46 @@ public ConsumerConfiguration setConsumerName(String consumerName) {
         this.consumerName = consumerName;
         return this;
     }
+
+    /**
+     * @return the maximum amount of messages to be sent in a single RedeliverUnacknowledgedMessages command,
+     * <i>(default: 1000)</i>
+     */
+    public int getMaxRedeliverUnacknowledgedMessagesBatch() {
+        return maxRedeliverUnacknowledgedMessagesBatch;
```

Comment:
On a second thought, instead of having a user-configurable setting, we could have an internal limit on how many messages to ask for redelivery with a single protobuf command. 

Eg.: each redelivery command can contain up to 1000 position. If there are more than 1000 timed out messages, the client will send multiple redelivery commands.

This would also solve the problem of an application setting that limit to big.